oak-parent/pom.xml

```diff
@@ -868,7 +828,7 @@
 
    <!--  - - - - - - - - - - - - - - - - - - - will run all the required tests before a release -->
    <profile>
-      <id>release</id>
```

Comment:
which profile has been used in the past? The one from here or the one from the reactor root?

Reply:
You mean while creating a release? I think the one in the reactor root. When I created the current 1.44.0 release I did not specify the profile `release`. I just ran the maven release commands without setting any profile explicitly.

This release profile was created with https://issues.apache.org/jira/browse/OAK-4136

I still don't quite understand why it was created, but I would rather refrain from changing the profile name to `apache-release`. I think it might mess up the release process.

Reply:
Ok, even the formerly used ASF Parent 18 had https://github.com/apache/maven-apache-parent/blob/aad3c292f623f798936bf8daeb5237a53282eaa4/pom.xml#L208. Therefore this profile seems to be legacy and can be removed.

Reply:
And just to properly close the loop here. Davide created the profile to run tests and checks before the actual release. The `release` profile is not intended to be used while creating the release.

Reply:
I will add a comment making the intent more clear. Then later on we can decide if we want to remove it.